runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearStream.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2018, Infinitay <https://github.com/Infinitay>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.tearsofguthix;
+
+import java.time.Duration;
+import java.time.Instant;
+import lombok.Getter;
+import net.runelite.api.DecorativeObject;
+
+public class TearStream
+{
+	@Getter
+	private Instant timer;
+
+	@Getter
+	private DecorativeObject streamDecorativeObject;
+
+	public TearStream(DecorativeObject streamDecorativeObject)
+	{
+		this.streamDecorativeObject = streamDecorativeObject;
+		timer = Instant.now();
+	}
+
+	public double getTimeLeft()
+	{
+		Duration duration = Duration.between(timer, Instant.now());
+		return 1 - (duration.compareTo(Duration.ofSeconds(9)) < 0 ? (double) duration.toMillis() / Duration.ofSeconds(9).toMillis() : 1);

```

Comment:
Probably make this Duration.ofSeconds(9) constant? also, this is probably not necessary at all, just compare duration secons to 9? maybe?

Reply:
I am mimicking the existing Hunting plugin. Also, if I do a basic compareTo then I can't have the "countdown effect" I have made it to a constant